test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -765,6 +766,159 @@ void testBoundsInference_5() {
   }
 }
 
+void testLoopNestComputeAt_1() {
+  // Verify that compute_at works on the following example:
+  //
+  // for (int i_a = 0; i_a < N; i_a++) {
+  //   A[i_a] = i_a * i_a
+  // }
+  // for (int i_b = 0; i_b < N; i_b++) {
+  //   B[i_b] = A[i_b]
+  // }
+  //
+  // After the transformation the i_b loop should have an allocation for a temp
+  // buffer and that buffer should be used in computation of B. No use of A
+  // should be in that loop after the transformation. Also, computation of A
+  // should not be inlined into B. Instead, it should be computed into the temp,
+  // and the temp should be used in B.
+  KernelScope kernel_scope;
+  VarHandle N("N", kInt);
+  Tensor* A = Compute(
+      "A", {{N, "i_a"}}, [&](const VarHandle& i_a) { return i_a * i_a; });
+  Tensor* B = Compute(
+      "B", {{N, "i_b"}}, [&](const VarHandle& i_b) { return A->call(i_b); });
+  LoopNest l({B});
+  std::vector<For*> loops = l.getLoopStmtsFor(B);
+  l.computeAt(l.getLoopBodyFor(A), loops[0]);
+  l.prepareForCodegen();
+  Stmt* s = l.root_stmt();
+
+  std::ostringstream oss;
+  oss << *s;
+
+  const std::string& verification_pattern =
```

Comment:
I am not a big fan of this string check pattern. It is difficult to strike a balance that checks enough, but not too much.

I would prefer manually construct the reference loop-nest, convert to string, and match two strings are equal.

Reply:
I actually much prefer this than byte-to-byte comparison os strings, which is way too fragile and will break any time we change something internally. Using patterns I can specify what exactly to check for: e.g. a compute at needs to insert `Allocate` of a given size: I can check for that and at the same time ignore, say, the temp variable name which can change at any point. Another example that can easily change in future: for temporary computations we currently replicate the entire original loopnest even if iteration over some of the axis has a range of 1. E.g. we can generate:
```
for y in 0..1:
  for x in 0..W:
     temp[y,x] = ...
```
At some point we will start optimizing out loops `for y in 0..1` - this would also break the test if I check for the exact match of strings.